meveo-admin/ejbs/src/main/java/org/meveo/service/base/BusinessService.java

```diff
@@ -182,82 +186,103 @@ public BusinessEntity findByEntityClassAndCode(Class<?> clazz, String code) {
      */
     @SuppressWarnings({ "rawtypes", "deprecation" })
 	public MeveoModule findModuleOf(P entity){
     	MeveoModule module = null;
     	if (entity != null) {
-    		Session session = this.getEntityManager().unwrap(Session.class);
-    		Query q = session.createQuery("SELECT m.module FROM ModuleItem m WHERE m.code = :code AND m.ItemType = :itemType");
-    		q.setString("code", entity.getCode());
-    		q.setString("itemType", entity.getClass().getName());
-    		module = (MeveoModule) q.getResultList().get(0);
-    	}
+    		if (entity instanceof CustomFieldTemplate) {
+    			CustomFieldTemplate entityCtf = (CustomFieldTemplate) entity;
+    			
+				Session session = this.getEntityManager().unwrap(Session.class);
+	    		Query q = session.createQuery("SELECT m.module FROM ModuleItem m WHERE m.code = :code AND m.itemType = :itemType AND m.applies_to = :applies_to");
+	    		q.setString("code", entityCtf.getCode());
+	    		q.setString("itemType", entityCtf.getClass().getName());
+	    		q.setString("applies_to", entityCtf.getAppliesTo());
+	    		if (!(q.getResultList().isEmpty())) {
+	    			module = (MeveoModule) q.getResultList().get(0);
+	    		}
+    		}else {
+				Session session = this.getEntityManager().unwrap(Session.class);
+	    		Query q = session.createQuery("SELECT m.module FROM ModuleItem m WHERE m.code = :code AND m.itemType = :itemType");
+	    		q.setString("code", entity.getCode());
+	    		q.setString("itemType", entity.getClass().getName());
+	    		if (!(q.getResultList().isEmpty())) {
+	    			module = (MeveoModule) q.getResultList().get(0);
+	    		}
+    		}
+		}
     	return module;
     }
     
     /**
      * Remove the entity that belongs to the module
      * 
-     * @param entity
-     * @param module
+     * @param entity belonging to the module
+     * @param module corresponding to the entity
+     * @throws BusinessException if the folder is not deleted
      */
-    public void removeFilesFromModule(P entity, MeveoModule module) {
-    	MeveoModuleDto meveoModuleDto = new MeveoModuleDto(module);
+    public void removeFilesFromModule(P entity, MeveoModule module) throws BusinessException {
     	BusinessEntityDto businessEntityDto = new BusinessEntityDto(entity);
-    	File fileToDelete = new File("..\\git\\"+meveoModuleDto.getCode()+"\\"+businessEntityDto.getClass().getAnnotation(ModuleItem.class).path()+"\\"+entity.getCode()+".json");
-
-    	fileToDelete.delete();
+    	
+    	File file = GitHelper.getRepositoryDir(currentUser, module.getGitRepository().getCode());
+    	String path = businessEntityDto.getClass().getAnnotation(ModuleItem.class).path() + "/" + entity.getCode();
+    	File removeFile = new File(file, path);
+    	if (removeFile.exists()) {
+    		try {
+				FileUtils.deleteDirectory(removeFile);
+			} catch (IOException e) {
+				throw new BusinessException("Folder unsuccessful deleted : " + removeFile.getPath() + ". " + e.getMessage());
```

Comment:
Use the constructor that takes the exception as parameter so we can keep the stack trace

Reply:
Also take care of identation